src/graders/fn-def-guard.arr

```diff
@@ -0,0 +1,95 @@
+import npm("pyret-lang", "../../src/arr/compiler/well-formed.arr") as WF
+import npm("pyret-lang", "../../src/arr/compiler/compile-structs.arr") as CS
+import npm("pyret-lang", "../../src/arr/compiler/ast-util.arr") as AU
+import file("../core.arr") as C
+import file("../grading.arr") as G
+import file("../grading-builders.arr") as GB
+import file("../common/ast.arr") as CA
+import file("../common/markdown.arr") as MD
+import ast as A
+
+include either
+include from C:
+  type Id
+end
+
+provide:
+  mk-def-guard,
+  data DefGuardBlock,
+  check-fun-defined as _check-fun-defined,
+  fmt-fun-def as _fmt-fun-def
+end
+
+data DefGuardBlock:
+  | parser-error(err :: CA.ParsePathErr) # this should've been caught by wf
+  | fn-not-defined(name :: String, arity :: Number)
+  | wrong-arity(name :: String, expected :: Number, actual :: Number)
+end
+
+fun check-fun-defined(
+  path :: String,
+  name :: String,
+  arity :: Number
+) -> Option<DefGuardBlock>:
+  cases (Either) CA.parse-path(path):
+    | left(err) => some(parser-error(err))
+    | right(ast) =>
+      ast-ended = AU.append-nothing-if-necessary(ast)
+      cases (A.Program) ast-ended:
+        | s-program(_, _, _, _, _, _, body) =>
+          cases (A.Expr) body:
+            | s-block(_, stmts) => find-def(stmts, name, arity)
+            | else => some(fn-not-defined(name, arity))
+          end
+      end
+  end
+end
+
+fun find-def(
+  stmts :: List<A.Expr>,
+  expected-name :: String,
+  expected-arity :: Number
+) -> Option<DefGuardBlock>:
+  cases (List) stmts:
+    | empty => some(fn-not-defined(expected-name, expected-arity))
+    | link(st, rest) =>
+      cases (A.Expr) st:
+        | s-fun(_, actual-name, _, args, _, _, _, _, _, _) =>
+          if actual-name == expected-name:
+            actual-arity = args.length()
+            if actual-arity == expected-arity: none
```

Comment:
nit: don't `none` on the same line the readability cost outweighs the extra 2 lines.